packages/puppeteer/tsconfig.json

```diff
@@ -18,7 +18,10 @@
     "noImplicitThis": true,
     "noImplicitReturns": true,
     "moduleResolution": "node",
-    "esModuleInterop": true
+    "esModuleInterop": true,
+    "paths": {
```

Comment:
Why was this necessary?

Reply:
Without this it uses the types provided by puppeteer which doesn't export `JSONArray, JSONObject, ElementHandle` but the types file does

Reply:
We use types that are not exported by puppeteer and use `@types/puppeteer`

Reply:
Where do the types come from? I still don't understand. Why does adding `paths` change anything? Isn't this default behavior?

Reply:
Don't we usually fix this using the `typesRoot` property?

```
"typeRoots": ["node_modules/@types"]
```

Reply:
From https://github.com/puppeteer/puppeteer/pull/8547:


> [BREAKING CHANGE] We've removed the ability to declare return types in type arguments for the following reasons:
> - Setting them will indubitably break auto-deduction.
> - You can just use `as ...` in TypeScript to coerce the correct type (given it makes sense).

Why do we have to hack TS rather than casting (`as ...`)?

Reply:
That `typeRoots` value is also the default. Specifying it does nothing.

Reply:
I've solved this without any hacks: https://github.com/dequelabs/axe-core-npm/pull/535/commits/b30f8a6a6de52bb6eb5c8aa6b23696eaccf4d529